noir-projects/aztec-nr/aztec/src/encrypted_logs/log_assembly_strategies/default_aes128/note.nr

```diff
@@ -0,0 +1,445 @@
+ * tag_from: INPUT as sender: AztecAddress
+ *
+ * Note-specific:
+ * derive_note_randomness_from_shared_secret: false
+ *
+ */
+
+/*
+ * LOG LAYOUT CHOICE:
+ *
+ * Short explanation:
+ * log = [tag, epk, header_ct=[[contract_address, ct_len], pkcs7_pad], ct=[[pt], pkcs7_pad], some bytes padding, some fields padding]
+ *
+ * Long explanation:
+ * tag: Field
+ * epk: [Field, u8]
+ * header_ct: [[u8; 32], [u8; 2], [u8; 16]]
+ * ct: [[u8; 2], [u8; x], [u8; y]]
+ *
+ * More precisely (in pseudocode):
+ *
+ * log = [
+ *     tag: Field,
+ *     Epk: Field,
+ *
+ *     le_bytes_31_to_fields(
+ *
+ *         log_bytes: [
+ *             eph_pk_sign: [u8; 1],
+ *
+ *             header_ciphertext: aes_encrypt(
+ *                 contract_address: [u8; 32],
+ *                 ct_length: [u8; 2],
+ *
+ *                 // the aes128_encrypt fn automatically inserts padding:
+ *                 header_pt_aes_padding: [u8; 14], // `16 - (input.len() % 16)`
+
+ *             ): [u8; 48],
+ *
+ *             ciphertext: aes_encrypt(
+ *                 final_pt: [
+ *                     pt: {
+ *                         note_bytes: {
+ *                             storage_slot: [u8; 32],
+ *                             note_type_id: [u8; 32],
+ *                             ...note: [u8; N * 32],
+ *                         }: [u8; N * 32 + 64],
+ *                     }: [u8; N * 32 + 64],
+
+ *                 ]: [u8; N * 32 + 64],
+ *
+ *                 // the aes128_encrypt fn automatically inserts padding:
+ *                 pt_aes_padding: [u8; 16 - ( (|pt_length| + |pt|) % 16 )]
+ *
+ *             ): [u8; |pt| + |pt_aes_padding|]
+ *                [u8; |ct|]
+ *
+ *             log_bytes_padding_to_mult_31: [u8; 31 * ceil((1 + 48 + |ct|)/31) - (1 + 48 + |ct|)],
+ *                                           [u8; p]
+ *
+ *         ]: [u8; 1 + 48 + |ct| + p]
+ *
+ *     ): [Field; (1 + 48 + |ct| + p) / 31]
+ *
+ *     log_fields_padding: [Field; PRIVATE_LOG_SIZE_IN_FIELDS - 2 - (1 + 48 + |ct| + p) / 31],
+ *
+ * ]: [Field; PRIVATE_LOG_SIZE_IN_FIELDS]
+ *
+ *
+ */
+
+/********************************************************/
+// Disgusting arithmetic on generics
+/********************************************************/
+
+// In this section, instead of initialising arrays with very complicated generic
+// arithmetic, such as:
+// let my_arr: [u8; (((PT + (16 - (PT % 16))) + 49) + ((((((PT + (16 - (PT % 16))) + 49) + 30) / 31) * 31) - ((PT + (16 - (PT % 16))) + 49)))] = [0; (((PT + (16 - (PT % 16))) + 49) + ((((((PT + (16 - (PT % 16))) + 49) + 30) / 31) * 31) - ((PT + (16 - (PT % 16))) + 49)))];
+//... we instead do the arithmetic a little bit at a time, so that the computation
+// can be audited and understood. Now, we can't do arithmetic on generics in the body
+// of a function, so we abusing functions in the following way:
+
+// |full_pt| = |pt| = (N * 32) + 64
+fn get_arr_of_size__full_plaintext<let PT: u32>() -> [u8; PT] {
+    [0; PT]
```

Comment:
You can use `std::mem::zeroed` in these bodies to remove the duplication

Reply:
Found it: this is what scares me away from ever using zeroed:
<img width="1128" alt="image" src="https://github.com/user-attachments/assets/30183ac7-3477-4d84-bcc7-182dfd6104fc" />

Reply:
Left a comment in the other bit about zeroed. I like zeroed.

Reply:
Oh no! I like `[0; PT]`. We have ourselves a Mexican standoff!!!
What happens next?

Reply:
I'll take the high road. I've changed it to zeroed.